src/components/upload/FileDiv.tsx

```diff
@@ -59,12 +95,86 @@ interface Props {
   onRemoveFile: (file: string) => any;
 }
 
-const FileDiv: FC<Props> = ({ hash, fileName, onRemoveFile }: Props) => {
-  const [hover, setHover] = useState(false);
+const FileDiv: FC<Props> = ({
+  hash,
+  fileName,
+  fileBlob,
+  onRemoveFile,
+}: Props) => {
+  const [hover, setHover] = useState<boolean>(false);
+  const [keywords, setKeywords] = useState<string[]>([]);
+  const [numArticles, setNumArticles] = useState<number>(0);
+  const [numReferences, setNumReferences] = useState<number>(0);
+  const [loadingProgress, setLoadingProgress] = useState<number>(80);
+
+  const getKeywordsList = (text: string) => {
+    const identifier = "ID ";
+    const keywordsLines = text
+      .split("\n")
+      .filter((line) => line.startsWith(identifier));
+    return keywordsLines
+      .map((line) =>
+        line
+          .replace(identifier, "")
+          .trim()
+          .split(";")
+          .map((keyword) => keyword.trim().toLowerCase())
+          .filter((keyword) => Boolean(keyword))
+      )
+      .flat();
+  };
+
+  const mostCommonKeywords = (text: string, max: number = 3) => {
+    const keywordsList = getKeywordsList(text);
+    let count: { [keyword: string]: number } = {};
+    for (let keyword of keywordsList) {
+      count[keyword] = (count[keyword] ? count[keyword] : 0) + 1;
+    }
+    const sortCount = Object.entries(count).sort((first, second) =>
+      first[1] < second[1] ? 1 : -1
+    );
+    return sortCount.slice(0, max).map((item) => item[0]);
+  };
+
+  const countArticles = (text: string) => {
+    const identifier = "PT ";
+    return text.split("\n").filter((line) => line.startsWith(identifier))
+      .length;
+  };
+
+  const countReferences = (text: string) => {
+    const identifier = "NR ";
+    return text
+      .split("\n")
+      .filter((line) => line.startsWith(identifier))
```

Comment:
I saw that you're using arrow functions with one parameter with parenthesis, in this specific cases, can be without parenthesis, we can start using prettier and configure it to automatically prettify on save in our editors, make sense?

Reply:
yes, my friend. Prettier (our formatter) wraps up the parameter in parenthesis, but we could remove them.

Reply:
This is the default prettier config.